### PR TITLE
Syscall filter fixes, tighter nspawn seccomp sandbox by default

### PR DESCRIPTION
**Pull request by poettering**
various syscall filter fixes

### CODE REVIEW

**Comment by poettering** [comment]
@topimiettinen please have a look!

**Comment by topimiettinen** [comment]
They look fine, but systemd-networkd seems to fail tests?

The groups could be useful for nspawn too.

**Comment by poettering** [comment]
yeah, the networkd thing is unrelated, should be fixed by #3499 hopefully...

File: TODO [hunk 1]
```diff
@@ -47,6 +47,10 @@ Features:
 
 * RestrictNamespaces= or so in services (taking away the ability to create namespaces, with setns, unshare, clone)
 
+* RestrictRealtime= which takes aware ability to create realtime processes
+
+* nspawn: make /proc/sys/net writable?
```
Comment (keszybz) [inline TODO:52]: I think you could just push those kind of things directly too master, unless they're somehow related to the PR.

File: src/shared/seccomp-util.c [hunk 2]
```diff
@@ -197,40 +227,26 @@ const SystemCallFilterSet syscall_filter_sets[] = {
                 "_sysctl\0"
                 "afs_syscall\0"
                 "break\0"
-                "fattach\0"
-                "fdetach\0"
+                "create_module\0"
                 "ftime\0"
                 "get_kernel_syms\0"
-                "get_mempolicy\0"
```
Comment (keszybz) [inline src/shared/seccomp-util.c:204]: This is removed, but is not added anywhere. If this is on purpose, the commit message should mention it. Similarly for vm86.

According to tuxcall(2), afs_syscall,  break,  fattach,  fdetach,  ftime,  getmsg, getpmsg, gtty, isastream, lock, madvise1, mpx, prof, profil, putmsg, putpmsg, security, stty, tuxcall, ulimit, vserver are not implemented. It's pretty hard to verify this list, but for the few I checked, it is accurate. We should probably remove all of them from this list then.

**Comment by keszybz** [comment]
LGTM. I think the commit message for the syscall list update should be improved though.

File: src/shared/seccomp-util.c (hunk 2 above)
Comment (poettering) [inline src/shared/seccomp-util.c:204]: vm86 is actually moved to `@cpu-emulation` and that's documented explicitly in the commit msg.

`get_mempolicy()` isn't obsolete, it's a NUMA thing, hence I removed it, but yeah this should have been mentioned in the commit msg.

So regarding the "not implemented" thing: I figured the syscalls that are not implemented but actually got a syscall nr assigned should be included in `@obsolete`, and the syscalls that ere neither implemented nor a syscall nr assigned should not.  `getpmsg()` is one of the latter, `getmsg()` is of the former, hence I removed `getmsg()` but left `getpmsg()` in there...

The documentation of `@obsolete` says explicitly that it masks non-implemented syscalls, hence I think it should really do that.

**Comment by poettering** [comment]
Will add the comment regarding mem_getpolicy() to the commit msg and then merge.

**Comment by poettering** [comment]
OK, merging as @keszybz liked it already, and I only made a change to the commit msg in my most recent push.

**Pull request merged**